subsys/bluetooth/audio/shell/bap.c

```diff
@@ -285,85 +276,65 @@ static void lc3_audio_timer_timeout(struct k_work *work)
 		return;
 	}
 
-	k_work_schedule(k_work_delayable_from_work(work),
-			K_USEC(txing_stream->qos->interval));
-
-	if (lc3_start_time == 0) {
-		/* Read start time and produce the number of frames needed to catch up with any
-		 * inaccuracies in the timer. by calculating the number of frames we should
-		 * have sent and compare to how many were actually sent.
-		 */
-		lc3_start_time = k_uptime_get();
-	}
-
-	uptime = k_uptime_get();
-	run_time_ms = uptime - lc3_start_time;
-
-	/* PDU count calculations done in 100us units to allow 7.5ms framelength in fixed-point */
-	run_time_100us = run_time_ms * 10;
-	sdu_goal_cnt = run_time_100us / (frame_duration_100us * frames_per_sdu);
-
-	/* Add primer value to ensure the controller do not run low on data due to jitter */
-	sdu_goal_cnt += prime_count;
-
-	if ((lc3_sdu_cnt % 100) == 0) {
-		shell_info(ctx_shell, "LC3 encode %d frames in %d SDUs\n",
-			   (sdu_goal_cnt - lc3_sdu_cnt) * frames_per_sdu,
-			   (sdu_goal_cnt - lc3_sdu_cnt));
-	}
-
 	seq_num = get_next_seq_num(txing_stream->qos->interval);
 
-	while (lc3_sdu_cnt < sdu_goal_cnt) {
-		const uint16_t tx_sdu_len = frames_per_sdu * octets_per_frame;
-		struct net_buf *buf;
-		uint8_t *net_buffer;
-		off_t offset = 0;
-		int err;
-
-		buf = net_buf_alloc(&sine_tx_pool, K_FOREVER);
-		net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+	const uint16_t tx_sdu_len = frames_per_sdu * octets_per_frame;
+	struct net_buf *buf;
+	uint8_t *net_buffer;
+	off_t offset = 0;
+	int err;
 
-		net_buffer = net_buf_tail(buf);
-		buf->len += tx_sdu_len;
+	buf = net_buf_alloc(&sine_tx_pool, K_FOREVER);
+	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
 
-		for (int i = 0; i < frames_per_sdu; i++) {
-			int lc3_ret;
+	net_buffer = net_buf_tail(buf);
+	buf->len += tx_sdu_len;
 
-			lc3_ret = lc3_encode(lc3_encoder, LC3_PCM_FORMAT_S16,
-					     audio_buf, 1, octets_per_frame,
-					     net_buffer + offset);
-			offset += octets_per_frame;
+	for (int i = 0; i < frames_per_sdu; i++) {
```

Comment:
It's still a shame that we have to rely on this for loop instead of fixing whatever is causing it to not work otherwise. 

But it is, in any case, still an improvement, and the removal of the loop can be done in a later patch